javascript/javascript2/week2/aliSharifi/script.js

```diff
@@ -0,0 +1,435 @@
+    if (e.target === modal) {
+      closeModal(modal);
+    }
+  });
+
+  // Close on Escape key
+  document.addEventListener("keydown", (e) => {
+    if (e.key === "Escape" && modal.style.display === "block") {
+      closeModal(modal);
+    }
+  });
+}
+
+//close modal
+function closeModal(modal) {
+  modal.style.display = "none";
+  document.body.style.overflow = "auto";
+}
+
+function setupModalEventListeners(modalContent, movieId) {
+  const stars = modalContent.querySelectorAll(".stars i");
+  const ratingDisplay = modalContent.querySelector(".rating-display");
+  const submitComment = modalContent.querySelector(".submit-comment");
+  const commentInput = modalContent.querySelector(".comment-input");
+  const commentsList = modalContent.querySelector(".comments-list");
+
+  // Load existing ratings and comments for this movie
+  const savedRating = localStorage.getItem(`movie-${movieId}-rating`);
+  const savedComments =
+    JSON.parse(localStorage.getItem(`movie-${movieId}-comments`)) || [];
+
+  // Set up rating system
+  if (savedRating) {
+    highlightStars(stars, parseInt(savedRating));
+    ratingDisplay.textContent = `Your rating: ${savedRating}/5`;
+  }
+
+  stars.forEach((star) => {
+    star.addEventListener("mouseover", () => {
+      highlightStars(stars, parseInt(star.dataset.rating));
+    });
+
+    star.addEventListener("mouseout", () => {
+      if (savedRating) {
+        highlightStars(stars, parseInt(savedRating));
+      } else {
+        clearStars(stars);
+      }
+    });
+
+    star.addEventListener("click", () => {
+      const rating = parseInt(star.dataset.rating);
+      localStorage.setItem(`movie-${movieId}-rating`, rating);
+      highlightStars(stars, rating);
+      ratingDisplay.textContent = `Your rating: ${rating}/5`;
+    });
+  });
+
+  // Load existing comments
+  savedComments.forEach((comment) => {
+    const commentElement = document.createElement("div");
+    commentElement.className = "comment";
+    commentElement.textContent = comment;
+    commentsList.appendChild(commentElement);
+  });
+
+  // Set up comment submission
+  submitComment.addEventListener("click", () => {
+    const commentText = commentInput.value.trim();
+    if (commentText) {
+      const commentElement = document.createElement("div");
+      commentElement.className = "comment";
+      commentElement.textContent = commentText;
+      commentsList.insertBefore(commentElement, commentsList.firstChild);
+
+      // Save comment
+      saveComment(movieId, commentText);
+
+      commentInput.value = "";
+    }
+  });
+}
+
+function highlightStars(stars, rating) {
+  stars.forEach((star) => {
+    const starRating = parseInt(star.dataset.rating);
+    if (starRating <= rating) {
+      star.classList.add("active");
+    } else {
+      star.classList.remove("active");
+    }
+  });
+}
+
+function clearStars(stars) {
+  stars.forEach((star) => star.classList.remove("active"));
+}
+
+// Initialize the page when DOM is loaded
```

Comment:
You don't need to wait for the `DOMContentLoaded` event since your `<script>` tag is at the end of the `<body>`.